### DISCUSSION TITLE
[BP#8] Proposer un traitement asynchrone lorsque c'est possible

### DISCUSSION THREAD

**Discussion by devcrafting** [cnumr/best-practices#200] [Bonnes pratiques] [answered]
Cela me semble frappé du bon sens d'un point de vue UX et technique, par contre, je n'arrive pas bien à saisir l'impact sur la sobriété.

Le côté libéré le serveur de présentation ne me paraît pas très cohérent ? Techniquement on a peut de chance qu'on ait un serveur qui attende un autre serveur (sauf si on est toujours dans l'exemple de l'exécution d'une requête SQL...mais il faudra bien toujours un serveur qui exécute cette requête)

Est-ce qu'on imagine que l'utilisateur va trainer sur le site à attendre que sa tâche se termine ? Et donc qu'il gâche des ressources ?

Bref je suis dubitatif et je me demande notamment comment le classer dans la nouvelle catégorisation des impacts, peut-être que je n'ai pas saisi certains éléments. Peut-être qu'un petit liftig suffirait ?

**Comment by ACTLEM** [comment]
Je ne sais pas si cela répond à tes interrogations, mais l'asynchrone permet:

- Un lissage de la charge du serveur responsable du traitement donc une mutualisation aisée donc moins serveurs (cf en synchrone il y a l'impératif de ne pas trop faire attendre l'utilisateur donc on n'a moins de libertés concernant la charge maximale)
- Un temps effectif de l'utilisation du service moindre car l'utilisateur n'a pas à se soucier de la fin de l'exécution

Ex: En passant sur un système de notification, une entreprise à réussi à diviser par 3 les ressources serveurs nécessaires

**Reply by devcrafting** [reply]
ok peut-être juste un re-wording alors :), ça me semble plus clair avec ton explication

**Reply by DocRoms** [reply]
Je suis d'accord avec @devcrafting sur un rewrite. Quand on vois le `tout passer en asynchrone quand c'est possible` on pense généralement aux `async` `defer` coté HTML pour le JS ;) 

Pourtant, la règle spécifie (dans sa description) qu'il s'agit de process lent entre le front et le serveur .

Coté JS **tout** passer en `async` c'est une bonne pratique de #webPerf, mais un sacré cache-misère : ça permet de cacher le poids du JS en accélérant le chargement de la page. On peut très bien avoir un site avec 500 JS déféré (et je sais de quoi je parle) qui d'affiche vite : Le problème est toujours là, on a trop de JS, mais il à été "caché" par l'asynchronise des scripts ;)

**Comment by ACTLEM** [comment, marked as answer]
Création de l'issue pour rewording: https://github.com/cnumr/best-practices/issues/268
